### ISSUE TITLE
Don't Highlight Anything When Cursor is Not on Word

### ISSUE THREAD

**Issue by alexdunn** [qstrahl/vim-matchmaker#18]
You can see in the below screenshot that although my cursor is not actually on any word at all, Matchmaker is highlighting (underlining) the word closest to the cursor.  

![screen shot 2015-01-03 at 18 36 56](https://cloud.githubusercontent.com/assets/1313618/5604542/dee9ccd8-9377-11e4-956a-aff1e6c48017.png)

It's very strange that anything should be highlighted when I'm not focusing on anything.  I find it much more of a distraction than it is helpful.  Could we highlight words only when the cursor is truly on top of them?  Or perhaps at least make this an option?

Thanks!

**Comment by qstrahl** [comment]
I agree. The reason for this is we just use `expand('<cword>')` which is
interpreted as the word closest to the cursor. I could put in a check of
some kind to assert that the character under the cursor is a keyword
character.

**Comment by matt1003** [comment]
There is a potential solution described on stack overflow:
http://stackoverflow.com/questions/15020143/vim-script-check-if-the-cursor-is-on-the-current-word

<pre>
let cInd = strridx(getline('.'), expand("<cword>"), col('.') - 1)
let isOnKeyword = (cInd >= 0 && (cInd + strlen(expand("<cword>"))) >= (col('.') - 1))
</pre>
